R/app_server.R

```diff
@@ -7,21 +7,44 @@ app_server <- function(session,input, output) {
 
   ## Get the data  - temporarily added data file in the project. this will mimic the packages final behaviour
 
-  # data_path <- here::here("analysis", "data", "derived_data", "emu_theses_with_text.csv")
-  # emu_theses <- readr::read_csv(data_path)
-  # usethis::use_data(emu_theses, overwrite = TRUE)
-  data("emu_theses")
-
+  data("emu_clean")
+  data("emu_metadata")
+  emu_theses <- dplyr::left_join(emu_clean, emu_metadata, by = "ID")
 
   # Update filters
 
-  shiny::updateSelectInput(session,"year", label = "Select year:",
-                    choices = c('ALL', emu_theses$graduation_year |>
-                                          as.character() |>
-                                          unique()|>
-                                          sort()
-                                ),
-                    selected='ALL')
+  shiny::updateSliderInput(session,"year", label = "Select year:", step = 1,
+                           min = c(emu_theses$graduation_year |>
+                                     as.character() |>
+                                     unique()|>
+                                     sort() |>
+                                     min()
+                           ),
+                           max = c(emu_theses$graduation_year |>
+                                     as.character() |>
+                                     unique()|>
+                                     sort() |>
+                                     max()
+                           ),
+                           value = c(emu_theses$graduation_year |>
+                                       as.character() |>
+                                       unique()|>
+                                       sort() |>
+                                       min(),
+                                     emu_theses$graduation_year |>
+                                       as.character() |>
+                                       unique() |>
+                                       sort() |>
+                                       median())
+                             )
+
```

Comment:
Can we simplify `min`, `max` and `value` like this (example on `min`)?

```r
min = min(emu_theses$graduation_year)
```

`min()` should work without the previous three lines.

Reply:
Thanks. I have introduced the suggested changes, please have a look

Reply:
Looks good, thanks!